Develop

Thekid2002 opened the pull request.

Thekid2002 committed:
SW-47 * Setup api

Thekid2002 committed:
SW-47 * Fixed worthless migrations

Thekid2002 committed:
Merge pull request #3 from Trimesteret/SW-47

Sw 47

Thekid2002 committed:
SW-65 * added authentication

Thekid2002 committed:
SW-65 * Trying me fucking hardest on authorization

Thekid2002 committed:
SW-65 * Jeg vil hellere dø end at fikse det her

Thekid2002 committed:
in progress

Thekid2002 committed:
SW-65 * Fixed some of the public private

Thekid2002 committed:
Merge branch 'SW-65' of https://github.com/Trimesteret/API into SW-65

Thekid2002 committed:
SW-65 * Fixed public private fields and dbcontext

Thekid2002 committed:
SW-65 * Look minor changes

Thekid2002 committed:
SW-65 * Fixed verifytoken to use exceptions

Thekid2002 committed:
SW-65 * Fixed exceptions

Thekid2002 committed:
Merge pull request #5 from Trimesteret/SW-65

Sw 65 * Viktor fixer

Thekid2002 committed:
Defining classes

Thekid2002 committed:
SW-103 * CreateItem start

Thekid2002 committed:
SW-103 * Fixed alot of services

Thekid2002 committed:
Merge pull request #6 from Trimesteret/SW-103

Sw 103

Thekid2002 committed:
SW-108 * Fixed issues with the itemtypes

Thekid2002 committed:
SW-108 * Added wine spec

Thekid2002 committed:
Merge pull request #7 from Trimesteret/SW-108

Sw 108

DaniekFjaes commented on Models/Authentication/Admin.cs:
```diff
@@ -0,0 +1,52 @@
+    public void EditUser(User user, string firstName, string lastName, int phone, string email, string password, int? phoneNumber)
+    {
+        Console.WriteLine(user.GetType());
+    }
```
Either an incomplete function or a misleading function name

Thekid2002 replied:
https://youtu.be/vrOO4wvBhRI?si=df3fZeFQhzCfkmNf&t=69 Kan jeg lige frem vælge?

ChristofferGamel committed:
API set up for getitems

DaniekFjaes commented on Models/Orders/CustomerPurchaseOrder.cs:
```diff
@@ -0,0 +1,22 @@
+    // Parameterless constructor for EF Core
+    protected CustomerPurchaseOrder()
+    {
+    }
```
Apparently, ef core doesn't approve :(

DaniekFjaes reviewed (changes requested):
A little birdie told me that this doesn't actually work, due to some issues with ef core, empty constructors and lists. So that should probably be fixed before we merge into master.

Thekid2002 committed:
SW-99 * Fixed model and migrations

Thekid2002 committed:
Merge pull request #9 from Trimesteret/SW-99

Sw 99

Thekid2002 committed:
SW-108 * Added the getSelf and editSelf

Thekid2002 committed:
SW-108 * Formatering

Thekid2002 committed:
Merge branch 'develop' into SW-108

Thekid2002 committed:
SW-108 * Renamed variables

Thekid2002 committed:
Merge pull request #10 from Trimesteret/SW-108

Sw 108

Commit:
SW-109 * Fixed merge conflict

Commit:
Merge branch 'develop' into SW-109

Thekid2002 committed:
Fixed wine constructor

Thekid2002 committed:
Merge pull request #12 from Trimesteret/HotfixWineConstructor

Fixed wine constructor

Commit:
Merge branch 'develop' into SW-109

Commit:
SW-109 * WIP Sorting wine on API instead of on client

Thekid2002 committed:
SW-109 * Fixed webshop

Thekid2002 committed:
Merge pull request #13 from Trimesteret/SW-109

Sw 109

Thekid2002 committed:
SW-100 * Added mass to item

Thekid2002 committed:
SW-116 * Fixed hashing and authorization is now atleast

Thekid2002 committed:
Merge pull request #15 from Trimesteret/SW-116

SW-116 * Fixed hashing and authorization is now atleast

ChristofferGamel committed:
setup for post + enum fixes, UNTESTED

Thekid2002 committed:
Merge branch 'develop' into SW-100

ChristofferGamel committed:
stash went wrong

Thekid2002 committed:
SW-117 * Added role authorization

Thekid2002 committed:
SW-117 * Added comment

Thekid2002 committed:
Merge pull request #16 from Trimesteret/SW-117

SW-117 * Added role authorization

Thekid2002 committed:
Merge branch 'develop' into SW-100

vjohns22 committed:
(INCOMPLETE) Does not work

Db does not understand what WineType is

Thekid2002 committed:
Merge pull request #14 from Trimesteret/SW-100

SW-100 * Added mass to item

ChristofferGamel committed:
post edit

ChristofferGamel committed:
Revert "post edit"

This reverts commit 7a6e379d641011f5e986ef6da9f76b7d7359cad9.

Thekid2002 committed:
SW-112 * Edit users and role fixes

ChristofferGamel committed:
ch-ch-ch-changes

Thekid2002 committed:
SW-11 * Changed ALOT

Thekid2002 committed:
Merge branch 'develop' into SW-111

Thekid2002 committed:
Update ItemController.cs

Thekid2002 committed:
SW-111 * Fixed merge conflict

Thekid2002 committed:
SW-112 * Finished edituser pages

Thekid2002 committed:
SW-112 * Fixed naming

ChristofferGamel committed:
changed values for price to flaot

ChristofferGamel committed:
double > float

Thekid2002 committed:
Merge pull request #17 from Trimesteret/SW-112

SW-112 * Edit users and role fixes

Commit:
Merge branch 'develop' into SW-111

Commit:
SW-111 * Made itemservice take all relevant parameters

Thekid2002 committed:
SW-101 * Changed some shit

ChristofferGamel committed:
Suppliers UI WIP (needs iteem association)

NiklasBH committed:
SW-118 * Changing Password Backend

Added frontend functionality for changing user password. Some of the logic behind the password changes does not quite work as intended.

Thekid2002 committed:
SW-101 * Finished shoppingcart

1

Thekid2002 committed:
SW-101 * Removed endpoint

Thekid2002 committed:
SW-111 * Added migrations

Thekid2002 committed:
SW-111 * Removed outcommented code

Thekid2002 committed:
Merge pull request #20 from Trimesteret/SW-111

Sw 111

Thekid2002 committed:
Merge branch 'develop' into SW-101

Thekid2002 committed:
Merge pull request #19 from Trimesteret/SW-101

Sw 101

Thekid2002 committed:
Small fixx

Thekid2002 committed:
Merge branch 'SW-101' into develop

ChristofferGamel committed:
Merge branch 'develop' into SW-121

Commit:
SW-123 * Edit item now work

vjohns22 committed:
Merge pull request #21 from Trimesteret/SW-123

SW-123 * Edit item now works

Thekid2002 committed:
SW-124 * Better? Better.

Thekid2002 committed:
Merge pull request #22 from Trimesteret/SW-124

SW-124 * Better? Better.

ChristofferGamel committed:
measuring stick

NiklasBH committed:
SW-118 * Fixed comments on changes

ChristofferGamel committed:
Merge branch 'develop' into SW-121

Thekid2002 committed:
SW-129 * Works more and more

MarcusBach44 committed:
SW-119 * Backend Order part1

MarcusBach44 committed:
Update InboundOrder.cs

Thekid2002 committed:
SW-129 * Tryed my best

ChristofferGamel committed:
SupplierId attr added to items + migrations

Thekid2002 committed:
SW-129 * Items

MarcusBach44 committed:
SW-119 * Now gets data from database

ChristofferGamel committed:
Working state, Creates duplicate items

Thekid2002 committed:
SW-129 * Save item

Thekid2002 committed:
Merge branch 'develop' into 118

vjohns22 committed:
Merge pull request #23 from Trimesteret/SW-129

Sw 129

Thekid2002 committed:
Merge branch 'develop' into SW-118

Thekid2002 committed:
SW-118 * Fixed change password

Thekid2002 committed:
Merge pull request #18 from Trimesteret/118

SW-118 * Backend functionality for changing user password

ChristofferGamel committed:
Working state, needs merging for suppliers and ass. items

Thekid2002 committed:
SW-134

Commit:
SW-132 * Url now gets posted to database

ChristofferGamel committed:
Merge branch 'develop' into SW-121

ChristofferGamel committed:
Merged develop into SW-121

Thekid2002 committed:
SW-132 * Fix

Thekid2002 committed:
SW-135 * Data seeding

ChristofferGamel committed:
Items correctly associated with supplier

ChristofferGamel committed:
Associations -> Relations

Thekid2002 committed:
Merge branch 'develop' into SW-134

Thekid2002 committed:
Merge pull request #24 from Trimesteret/SW-134

SW-134

MarcusBach44 committed:
SW-119 * fixed comments maybe

Thekid2002 committed:
SW-131 * Begining enums

MarcusBach44 committed:
Update PurchaseOrder.cs

ChristofferGamel committed:
Edit Supplier Functionality Added

ChristofferGamel committed:
Merge branch 'develop' into SW-121

MarcusBach44 committed:
Merge branch 'develop' into SW-119

ChristofferGamel committed:
cleanup

MarcusBach44 committed:
SW-119 * Fixed merge conflicts

Thekid2002 committed:
SW-121 * Arbejder på det

Thekid2002 committed:
SW-121 * Fixed suppliers

Thekid2002 committed:
SW-131 * Edit enum finished

Thekid2002 committed:
SW-121 * Kommentare

ChristofferGamel committed:
Merge pull request #26 from Trimesteret/SW-121

Sw 121

Thekid2002 committed:
Merge branch 'develop' into SW-131

Thekid2002 committed:
Merge pull request #27 from Trimesteret/SW-131

Sw 131

Thekid2002 committed:
SW-119 * Purchase order and inbound order in progress

Thekid2002 committed:
Merge branch 'develop' into SW-119

Thekid2002 committed:
SW-119 * Et yderst jammerligt forsøg på at redde ordre

Thekid2002 committed:
SW-119 * Fixing orders

Thekid2002 committed:
SW-119 * Fix

Thekid2002 committed:
SW-119 * Fixing purchase order

Thekid2002 committed:
SW-119 * Fix

Thekid2002 committed:
SW-119 * SW-119

Thekid2002 committed:
SW-119 * Risalamande

Thekid2002 committed:
SW-119 * Så for satan

Thekid2002 committed:
Merge pull request #25 from Trimesteret/SW-119

Sw 119 * Created order page in warehouse, an gets orders from backend

Thekid2002 committed:
SW-174 * Unit test..ing my patience

Thekid2002 committed:
SW-174 * Added unit testing

Thekid2002 committed:
SW-174 * TESTING :pig:

Thekid2002 committed:
SW-174 * Test

Thekid2002 committed:
Begyndelse på nogle item tests

DaniekFjaes committed:
Invading your branch for fun and profit

Added testing of 2 branches for EditItem

DaniekFjaes committed:
Lil hashing test

Approved af Viktor og Andreas, så hvis det er dumt er det ikke min skyld.

Thekid2002 committed:
SW-174 * Testing and cleaning

Thekid2002 committed:
SW-174 * Testing

Thekid2002 committed:
Merge pull request #28 from Trimesteret/SW-174

Sw 174

Thekid2002 committed:
SW-178 * Fixed inbound orders

Thekid2002 committed:
SW-178 * Commit service and shit

DaniekFjaes committed:
Tests and fixes

DaniekFjaes committed:
Tests +

- Added new files to split tests up
- Covered most of PurchaseOrder functions in OrderService

DaniekFjaes committed:
Added comments to tests

Thekid2002 committed:
SW-178 * Removed html and css

Thekid2002 committed:
SW-178 * Tests inbound orders

Thekid2002 committed:
Merge branch 'SW-179' into SW-178

Thekid2002 committed:
SW-178 * Remove shit

Thekid2002 committed:
SW-178 * Test

DaniekFjaes committed:
Merge pull request #30 from Trimesteret/SW-178

Sw 178

Thekid2002 committed:
Fixed testing
